src/Farmer/Builders/Builders.WebApp.fs

```diff
@@ -466,23 +464,23 @@ type WebAppConfig =
 
             for (ExtensionName extension) in this.SiteExtensions do
                 { Name = ResourceName extension
-                  SiteName = this.Name
+                  SiteName = this.Name.ResourceName
                   Location = location }
 
             if Map.isEmpty this.CommonWebConfig.Slots then
                 site
             else
                 { site with AppSettings = Map.empty; ConnectionStrings = Map.empty }
                 for (_,slot) in this.CommonWebConfig.Slots |> Map.toSeq do
-                    slot.ToArm site
+                    slot.ToSite site
 
             yield! (PrivateEndpoint.create location this.ResourceId ["sites"] this.PrivateEndpoints)
         ]
 
 type WebAppBuilder() =
     member __.Yield _ =
         { CommonWebConfig = 
-            { Name = ResourceName.Empty
+            { Name = WebAppName.Create("defaultvalue").OkValue
```

Comment:
I was thinking about this. The problem is the the `WebAppName` of course doesn't allow an empty name ! :-) I copied the same pattern that I used with (I think) SQL Server.

What I could do it make `defaultvalue` a magic string and throw an error (or warning?) on the `Run` method of the builder?

Reply:
Only when we're using `ResourceName`. Storage and SQL both have validation and suffer from this same issue. We could also make the field optional, but this would then bleed downstream which I really wouldn't want.

Let me try a couple of options out and see what they look like and get back to you.

Reply:
I've tried an Option variant out - it's too painful to let it bleed out - it's used by Functions, EventGrid, Traffic Manager etc. - by the time we get to them, the optionality should not be there. So that's one possibility we can rule out.

Reply:
Ok, I've tried a generic WebAppConfig which could be specialised by what the Name type was e.g. ResourceName or WebAppConfig. This doesn't work either because members can't be bound to generic specializations.

Reply:
@ninjarobot if you're cool with this "as-is", please feel free to merge whenever you like - I'd rather not accept my own PR ;-)